server/api/products.js

```diff
@@ -23,7 +21,7 @@ router.get('/', (req, res, next) => {
     });
 });
 router.get('/:id', (req, res, next) => {
-  const id = parseInt(req.params.id);
+  // const id = parseInt(req.params.id);
```

Comment:
you should validate that 'id' is in the right format.
so if id should be a number you should check that it is in fact a number before making the query